server/src/services/library.service.ts

```diff
@@ -540,6 +533,34 @@ export class LibraryService implements OnEvents {
     return JobStatus.SUCCESS;
   }
 
+  // Checks if an online asset should be marked as offline, either due to missing path, or path outside of any import path
+  async handleOfflineCheck(job: ILibraryOfflineJob): Promise<JobStatus> {
+    const asset = await this.assetRepository.getById(job.id);
+
+    if (!asset || asset.isOffline) {
+      // We only care about online assets, we exit here if offline
+      return JobStatus.SKIPPED;
+    }
+
+    const exists = await this.storageRepository.checkFileExists(asset.originalPath, R_OK);
+
+    if (exists) {
+      const isInPath = job.importPaths.find((path) => asset.originalPath.startsWith(path));
+
+      if (isInPath) {
+        // Asset path exists and is within an import path
+        this.logger.verbose(`Asset is still online: ${asset.originalPath}`);
+        return JobStatus.SUCCESS;
+      }
+    }
+
+    // Either asset path does not exist, or it is outside of any import path
+    this.logger.debug(`Marking asset as offline: ${asset.originalPath}`);
+    await this.assetRepository.update({ id: asset.id, isOffline: true });
+
+    return JobStatus.SUCCESS;
+  }
+
```

Comment:
I think a lot of these comments are unnecessary. If anything some could be communicated with better variable names.

Reply:
Done